src/kinetics/RxnRates.cpp

```diff
@@ -90,21 +90,30 @@ Plog::Plog(const std::multimap<double, Arrhenius>& rates)
 
 void Plog::validate(const std::string& equation)
 {
+    fmt::memory_buffer err_reactions;
     double T[] = {200.0, 500.0, 1000.0, 2000.0, 5000.0, 10000.0};
-    for (auto iter = pressures_.begin(); iter->first < 1000; iter++) {
+
+    for (auto iter = ++pressures_.begin(); iter->first < 1000; iter++) {
         update_C(&iter->first);
         for (size_t i=0; i < 6; i++) {
-            double k = updateRC(log(T[i]), 1.0/T[i]);
+            double k = 0;
+            for (size_t p = ilow1_; p < ilow2_; p++) {
+                k += rates_[p].updateRC(log(T[i]), 1.0/T[i]);
+            }
+            // avoid to add invalid k at cantera inserted pressure(P=0) to error message
             if (!(k >= 0)) {
                 // k is NaN. Increment the iterator so that the error
                 // message will correctly indicate that the problematic rate
                 // expression is at the higher of the adjacent pressures.
-                throw CanteraError("Plog::validate",
-                    "Invalid rate coefficient for reaction '{}'\nat P = {}, T = {}",
-                    equation, std::exp((++iter)->first), T[i]);
+                format_to(err_reactions, "\nInvalid rate coefficient for reaction '{}'\nat P = {:.5g} , T = {}",
+                          equation, std::exp((iter)->first), T[i]);
             }
         }
     }
+    if (err_reactions.size()) {
+        throw CanteraError("Plog::validate",
+                           "\n{}", to_string(err_reactions));
```

Comment:
I think the formatted error message ends up containing more blank lines than it needs to, which you can improve by removing the one that's added on this line. The `CanteraError` constructor is a bit more clever than `format_to`, so you can just remove the argument `"\n{}"` and get the behavior you want.